firstapp/app/src/androidTest/java/com/example/firstapp/UtilityMethods.java

```diff
@@ -26,13 +36,28 @@ static UiDevice launchApp(String pkg, int timeout) {
         // Launch the app
         Context context = ApplicationProvider.getApplicationContext();
         final Intent intent = context.getPackageManager().getLaunchIntentForPackage(pkg);
+
         // Clear out any previous instances
         intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK);
+
+        //startActivity imposes a 5 second cool-down after the home button is pressed, so we wait
+        //out that cool-down before grabbing the time and launching
+        sleep(5000);
+        long start = SystemClock.elapsedRealtime();
         context.startActivity(intent);
 
         // Wait for the app to appear
         device.wait(Until.hasObject(By.pkg(pkg).depth(0)), timeout);
-        return device;
 
+        return start;
+
+    }
+
+
+    static void force_quit_app(String pkg) throws IOException, InterruptedException {
+        Runtime.getRuntime().exec(new String[] {"am", "force-stop", pkg});
+        sleep(5000);
```

Comment:
Why the sleep here?  Looks a bit random to sleep 5s and may add unnecessary delay.

Reply:
When resolving comments, should also provide a short comment explaining how it was resolved.